app/services/analytics/ecf_school_service.rb

```diff
@@ -32,14 +34,14 @@ def update_school_analytics
                 induction_tutor_nominated: school.induction_tutor.present?,
                 tutor_nominated_time: school.induction_tutor&.created_at,
                 induction_tutor_signed_in: school.induction_tutor&.current_sign_in_at&.present?,
-                induction_programme_choice: school.school_cohorts.first&.induction_programme_choice,
+                induction_programme_choice: school.school_cohorts.select { |sc| sc.cohort_id = current_cohort_id }&.first&.induction_programme_choice,
                 in_partnership: school.active_partnerships.any?,
                 partnership_time: school.active_partnerships.first&.created_at,
                 partnership_challenge_reason: school.active_partnerships.any? ? nil : school.partnerships.first&.challenge_reason,
                 partnership_challenge_time: school.active_partnerships.any? ? nil : school.partnerships.first&.challenged_at,
-                lead_provider: school.active_partnerships.first&.lead_provider&.name,
-                delivery_partner: school.active_partnerships.first&.delivery_partner&.name,
-                chosen_cip: school.school_cohorts.first&.core_induction_programme&.name,
+                lead_provider: school.active_partnerships.first&.lead_provider&.name || school.partnerships.first&.lead_provider&.name,
+                delivery_partner: school.active_partnerships.first&.delivery_partner&.name || school.partnerships.first&.delivery_partner&.name,
```

Comment:
This seems like a little extra - are we sure we want to surface the inactive partnerships?

Reply:
Specific request from Chris